grails-app/controllers/org/pih/warehouse/core/ErrorsController.groovy

```diff
@@ -98,7 +98,7 @@ class ErrorsController {
             response.status = 400
             BeanPropertyBindingResult errors = request?.exception?.cause?.errors
             def errorMessages = errors.allErrors.collect {
-                return messageService.getMessage(it.codes[0], it.arguments, (it.defaultMessage ?: it.codes[0]), localizationService.getCurrentLocale())
+                return g.message(error: it, args: it.arguments, default: it.defaultMessage, locale: localizationService.getCurrentLocale())
```

Comment:
If we don't want to cause any regression related to this, we can always add some additional arguments and do an `if` whether to return what you did, or the `messageService` thing.
Moreover - I think you could just simply catch the exception yourself, just doing `try/catch` inside `editProductSynonym` action, so you would not be dependent on this handler.

If we want to proceed with this change, we should at the very least change the `default` from:
```
default: it.defaultMessage
```
to what it was:
```
default: (it.defaultMessage ?: it.codes[0])
```

Reply:
Yes I am a bit wary of any potential regressions,  so I guess the safest option for me in this situation is to just do the `g.message(error: e.error )` inside of a catch block for edit product synonym